### DISCUSSION TITLE
New Idea: Combination And ID Locked Safe

### DISCUSSION THREAD

**Discussion by Githubberclown** [FakeFishGames/Barotrauma#6840] [Ideas and suggestions]
A Craftable Safe That When Placed Has A 3x3 Space and can hold anything as heavy as a riot shotgun, and can only be placed inside the submarine.

1: Combination lock that can be set by the person who set the combination lock in the first place (Example: Mark The Assistant Set The Lock And Can Access It Whenever He Likes) And Can Bypass The Lock Because He Set The Combination

2: This Lock Can Be Bypassed By Certain ID Cards Based On Captains Choice, Starting With The Priority System Labeled On The Side With A Glowing Tinge Around The Button To Let You Know They Can Bypass The Locker Combination With Their Own ID.

3: This Locker Can Be Bought And Crafted, As A Rough Example 2 Titanium Aluminium 2 FPGA And 1 Plastic

What Is The Purpose?: Extra Inventory Space For Individual People While Also Giving Way To Being Searched If People Are Hiding Potentially Valuable Items From The Host To Greif Or Damage.

**Comment by Githubberclown** [comment]
The Purpose Of This Is Also Meant To Be Mostly For A Single Person, Not Everyone With A Job ID Card

**Comment by StruggleDuck** [comment]
I do believe that something like this is already achievable in vanilla, using the in game sub editor. I can think of 2 possible ways to do this. 1) have as many secure cabinets as you need for your crew. Then, assign them a specific ID tag that they are required to have to open it. Finally, have a secure cabinet in the bridge with an ID card for each of the cabinets, with each one having its own unique ID tag, which corresponds to each locker. 

2) make it so that there is multiple spawns for each class, and have it so that only one person can spawn per spawn point. Then, essentially repeat the same steps as idea # 1, but this time instead of a secure cabinet with the id cards, just apply the ID tag to the spawn points directly. This way the crew should have their own locker, which can be opened using their regular ID card.

Im not entirely sure if idea # 2 can actually work, as I am unsure if you can make it so only one person can spawn per spawn point.
